### DISCUSSION TITLE
A way to get `className` or `classList` of the parsed element in the rule context?

### DISCUSSION THREAD

**Discussion by lunelson** [unocss/unocss#1030] [Q&A] [answered]
I'd like to implement a rule which outputs CSS specific to the selector _as well as some "default" or "base" styles_, as long as those defaults are not already set or implied by another selector on the same element. Is there a way I can get access to the className string or the classList (array of strings) for the parsed element (as part of the `RuleContext` that is passed to the output function), so that I could check for this condition?

**Comment by chu121su12** [comment, marked as answer]
In short, rules work on individual token obtained from extractor that splits incoming strings.

I guess your best bet is to implement different extractor and then using custom shortcut to detect/process the whole opening tag/raw classList.

**Reply by lunelson** [reply]
OK I see. My mental-model of this so far was imagining something more complex, but it sounds like the extractor is pulling out className values from code-as-string rather than walking through the AST of the code, is that right? This would mean there's no chance to back-trace what the coincident classNames on any given node might be, so I'd have to come at this a different way
